File: Python3.x/27-Remove Element.py
# Runtime: 16 ms, faster than 95.57% of Python online submissions for Remove Element.
# Memory Usage: 11.9 MB, less than 13.53% of Python online submissions for Remove Element.
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        i = 0
        for j, _ in enumerate(nums):
            if nums[j] != val:
                nums[i] = nums[j]
                i += 1
        return i

File: Algorithms/C++/27-Remove_Element.cpp
// Runtime: 4 ms, faster than 70.71% of C++ online submissions for Remove Element.
// Memory Usage: 8.5 MB, less than 97.06% of C++ online submissions for Remove Element.
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int j = 0;
        for (int i = 0; i < nums.size(); ++i)
            if (nums[i] != val) {
                nums[j] = nums[i];
                ++j;
            }
        return j;
    }
};
